### PR TITLE
fix: pass mock context when resetting the DB

### PR DESCRIPTION
**Pull request by spaenleh**

### CODE REVIEW

File: src/mockServer/msw/handlers.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ const getMemberIdFromToken = (bearer: string | null): string => {
 export const buildMSWMocks = (
   { apiHost }: LocalContext,
   database?: Database,
+  appContext?: LocalContext,
```
Comment (ReidyT) [inline src/mockServer/msw/handlers.ts:49]: Wouldn't it be clearer if buildMSWMocks ask apiHost and appContext without apiHost ?
It avoids a possible confusion when seeing something like `buildMSWMocks(context, db, context, 'db name');`.

With these changes: 
```
const buildMSWMocks = (
  apiHost: string,
  database?: Database,
  appContext?: Omit<LocalContext, 'apiHost'>,
  dbName?: string,
)
```

you can then call the function with less confusion ? `buildMSWMocks('host', db, context, 'db name');`

If I missed or misunderstand something, just pass this comment.

File: src/mockServer/msw/server.ts [hunk 2]
```diff
@@ -17,7 +17,7 @@ export const mockServiceWorkerServer = ({
   dbName?: string;
 }): { worker: SetupWorker; resetDB: (data: Database) => void } => {
   const fullAppContext = buildMockLocalContext(appContext);
-  const mswMocks = buildMSWMocks(fullAppContext, database, dbName);
+  const mswMocks = buildMSWMocks(fullAppContext, database, fullAppContext, dbName);
```
Comment (ReidyT) [inline src/mockServer/msw/server.ts:20]: I think using the same variable twice add confusion. My comment about the buildMSWMocks could maybe solve this.

**Review by ReidyT — APPROVED** [review]
If the duplication of LocalContext in buildMSWMocks is wanted, then this PR seems to be ok. Otherwise, I would suggest to ask `apiHost` explicitly in the `buildMSWMocks` function.

**Comment by spaenleh** [comment]
@ReidyT I had missed the duplication. Fixed in the new commits. Thank you for noticing 👍

**Pull request merged**